pkg/ebpf/c/tracee.bpf.c

```diff
@@ -555,32 +557,54 @@ int tracepoint__sched__sched_process_fork(struct bpf_raw_tracepoint_args *ctx)
         }
     }
 
+    // TODO: decide if SCHED_PROCESS_FORK signal should be always submitted or just for traced evts
     if (!should_trace(&p))
         return 0;
 
     // follow every pid that passed the should_trace() checks (used by the follow filter)
     c_proc_info->follow_in_scopes = p.task_info->matched_scopes;
 
-    if (should_submit(SCHED_PROCESS_FORK, p.event) || p.config->options & OPT_PROCESS_INFO) {
-        int parent_ns_pid = get_task_ns_pid(parent);
-        int parent_ns_tgid = get_task_ns_tgid(parent);
-        int child_ns_pid = get_task_ns_pid(child);
-        int child_ns_tgid = get_task_ns_tgid(child);
-
-        save_to_submit_buf(&p.event->args_buf, (void *) &parent_pid, sizeof(int), 0);
-        save_to_submit_buf(&p.event->args_buf, (void *) &parent_ns_pid, sizeof(int), 1);
-        save_to_submit_buf(&p.event->args_buf, (void *) &parent_tgid, sizeof(int), 2);
-        save_to_submit_buf(&p.event->args_buf, (void *) &parent_ns_tgid, sizeof(int), 3);
-        save_to_submit_buf(&p.event->args_buf, (void *) &child_pid, sizeof(int), 4);
-        save_to_submit_buf(&p.event->args_buf, (void *) &child_ns_pid, sizeof(int), 5);
-        save_to_submit_buf(&p.event->args_buf, (void *) &child_tgid, sizeof(int), 6);
-        save_to_submit_buf(&p.event->args_buf, (void *) &child_ns_tgid, sizeof(int), 7);
-        save_to_submit_buf(&p.event->args_buf, (void *) &start_time, sizeof(u64), 8);
+    int parent_ns_pid = get_task_ns_pid(parent);
+    int parent_ns_tgid = get_task_ns_tgid(parent);
+    int child_ns_pid = get_task_ns_pid(child);
+    int child_ns_tgid = get_task_ns_tgid(child);
+
+    // parent
+    save_to_submit_buf(&p.event->args_buf, (void *) &parent_pid, sizeof(int), 0);
+    save_to_submit_buf(&p.event->args_buf, (void *) &parent_ns_pid, sizeof(int), 1);
+    save_to_submit_buf(&p.event->args_buf, (void *) &parent_tgid, sizeof(int), 2);
+    save_to_submit_buf(&p.event->args_buf, (void *) &parent_ns_tgid, sizeof(int), 3);
+    // child
+    save_to_submit_buf(&p.event->args_buf, (void *) &child_pid, sizeof(int), 4);
+    save_to_submit_buf(&p.event->args_buf, (void *) &child_ns_pid, sizeof(int), 5);
+    save_to_submit_buf(&p.event->args_buf, (void *) &child_tgid, sizeof(int), 6);
+    save_to_submit_buf(&p.event->args_buf, (void *) &child_ns_tgid, sizeof(int), 7);
+    // start times
+    save_to_submit_buf(&p.event->args_buf, (void *) &task_start_time, sizeof(u64), 8);
+
+    // Submit the regular SCHED_PROCESS_FORK event
 
+    if (should_submit(SCHED_PROCESS_FORK, p.event) || p.config->options & OPT_PROCESS_INFO) {
         events_perf_submit(&p, SCHED_PROCESS_FORK, 0);
     }
 
-    return 0;
+    //
+    // Submit SCHED_PROCESS_FORK signal event now (only args, other perfbuffer)
+    //
+
+    // args: parent_start_time used by control plane ONLY
```

Comment:
Ok so these are unique only to the control plane versions of all these events/signals. So we should separate the concept of the signal into an explicit definition, since documenting it like this is a recipe for mistakes later down the line. 
I believe a minimal version of the event definition code would suffice, probably with no concurrency control but with encapsulation to ensure only reading is possible.